### PR TITLE
feat(cli): Added url hash to differentiate between urls

### PR DESCRIPTION
**Pull request by jaspritt-hora**
### ADDED:
- `hash` parameter, while generating reports for apps using #/ 
- Test case added for the same

### NOTE: 
- This change is in reference to [this](https://github.com/GoogleChrome/lighthouse-ci/discussions/834) thread, as per discussion with @patrickhulce.

### CODE REVIEW

**Comment by jaspritt-hora** [comment]
@patrickhulce - Changes are done. Let me know, please review and let me know, if anything is required.

File: packages/cli/src/upload/upload.js [hunk 1]
```diff
@@ -542,7 +542,7 @@ async function runFilesystemTarget(options) {
     const fetchTimeDate = new Date(new Date(lhr.fetchTime).getTime() || Date.now());
     const context = {
       hostname: url.hostname,
-      pathname: url.pathname,
+      pathname: url.hash + url.pathname,
```
Comment (patrickhulce) [inline packages/cli/src/upload/upload.js:545]: we don't want to merge these two, we would want them to be separate
```suggestion
      pathname: url.pathname,
      hash: url.hash,
```

**Review by patrickhulce — CHANGES_REQUESTED** [review]
thanks @jaspritt-hora!

could you add a test case here?

File: packages/cli/src/upload/upload.js (hunk 1 above)
Comment (jaspritt-hora) [reply packages/cli/src/upload/upload.js:545]: @patrickhulce - Regarding the test case, we want to add the test case for both `collect` & `upload` or just the latter?

Comment (patrickhulce) [reply packages/cli/src/upload/upload.js:545]: Collect isn't being touched here, just upload

Comment (jaspritt-hora) [reply packages/cli/src/upload/upload.js:545]: @patrickhulce - Sounds good! I have made the suggested change and written the test case in a new file and it is working as expected. 

However, I have a question, NOT all the test cases are running on my machine. 
Refer following screenshot - 

The unit test case file, I have written is - `upload-url-hash.test.js`. The test for this has passed. 
image

However, `NOT ALL` the test suites are passing. Refer following screenshots - 
image

Can I go ahead and commit the changes ? 
OR 
Do I need the entire test suite to pass?

Comment (patrickhulce) [reply packages/cli/src/upload/upload.js]: don't worry about the whole test suite for now, some are flaky and/or require other setup on your machine like mysql, if upload tests are passing you should be a-ok

Comment (jaspritt-hora) [reply packages/cli/src/upload/upload.js]: Perfect!

Comment (jaspritt-hora) [reply packages/cli/src/upload/upload.js]: @patrickhulce - Can you the checks again. The issue was related to syncing of the directories, I have added `rimraf` and the required code for the same. The test case is running successfully now, had 3 consecutive successful runs.

Comment (jaspritt-hora) [reply packages/cli/src/upload/upload.js]: @patrickhulce - Checked the raw logs for both `windows` and `macOS` checks that have failed, in that - the test case written for the PR has passed successfully.

The test case file name is - `upload-url-hash.test.js`.
Let me know, if there anything actionable at my end.

**Review by patrickhulce — APPROVED** [review]

**Pull request merged**